contracts/Vault1.sol

```diff
@@ -1,6 +1,78 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+// import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+// import "./SafeERC20.sol";
+
+// @title Vault1 
+// @notice a vault where users can deposit and withdraw any ERC20 token.
 contract Vault1 {
+
+  event newDeposit(uint _amount);
+  event newTransfer(uint _amount);
+
+  // token contract will hold
+  IERC20 private immutable _holdToken;
+
+  // who can withdraw
+  address private immutable _owner;
+
+  // @dev to receive deposit in the deploy transaction add "payable"
+  constructor (address _canWithdraw, IERC20 _token)  {
+    _owner = _canWithdraw;
```

Comment:
Anyone who deposits those tokens can withdraw. You must maintain a mapping of who deposited how much and allow them to withdraw if they are withdrawing less than or equal to what they deposited